src/Illuminate/Translation/Translator.php

```diff
@@ -157,13 +125,25 @@ public function getFromJson($key, array $replace = [], $locale = null)
         // using the typical translation file. This way developers can always just use a
         // helper such as __ instead of having to pick between trans or __ with views.
         if (! isset($line)) {
-            $fallback = $this->get($key, $replace, $locale);
-
-            if ($fallback !== $key) {
-                return $fallback;
+            [$namespace, $group, $item] = $this->parseKey($key);
+
+            // Here we will get the locale that should be used for the language line. If one
+            // was not passed, we will use the default locales which was given to us when
+            // the translator was instantiated. Then, we can load the lines and return.
+            $locales = $fallback ? $this->localeArray($locale) : [$locale];
+
+            foreach ($locales as $locale) {
+                if (! is_null($line = $this->getLine(
+                    $namespace, $group, $locale, $item, $replace
+                ))) {
+                    break;
```

Comment:
You can return `return $line ?: $key;` here because `makeReplacements` already ran in `getLine()` no need to run it again.

Reply:
I don't think so. If I do so then this test fails:

```php
    public function testGetJsonForNonExistingReturnsSameKeyAndReplaces()
    {
        $t = new Translator($this->getLoader(), 'en');
        $t->getLoader()->shouldReceive('load')->once()->with('en', '*', '*')->andReturn([]);
        $t->getLoader()->shouldReceive('load')->once()->with('en', 'foo :message', '*')->andReturn([]);
        $this->assertEquals('foo baz', $t->get('foo :message', ['message' => 'baz']));
    }
```

Reply:
It doesn't fail. `getLine` does the replacements already.

Reply:
Updated 👍